### PR TITLE
feat: add update action/command

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

File: mergify_engine/actions/update.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    validator = {}
+
+    @staticmethod
+    def run(pull, sources, missing_conditions):
+        try:
+            branch_updater.update_with_api(pull)
+        except branch_updater.BranchUpdateFailure as e:
+            return "failure", "Branch update failed", str(e)
+        else:
+            return "success", "Branch has been successfully update", ""
```
[the remark this answers is not in the document]
Comment (sileht) [reply mergify_engine/actions/update.py]: done

File: doc/source/actions.rst [hunk 2]
```diff
@@ -378,3 +378,8 @@ The ``rebase`` action will rebase the pull request against its base branch.
 
    Be aware that rebasing force-pushes the pull request head branch: any change
    done to the that branch while Mergify is rebasing will be lost.
+
+update
+======
+
+The ``update`` action update the pull request against its base branch.
```
Comment (jd) [inline doc/source/actions.rst]:
```suggestion
The ``update`` action updates the pull request against its base branch.
```

File: mergify_engine/actions/update.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# -*- encoding: utf-8 -*-
+#
+#  Copyright © 2019 Mehdi Abaakouk <sileht@sileht.net>
```
Comment (jd) [inline mergify_engine/actions/update.py]:
```suggestion
#  Copyright © 2020 Mehdi Abaakouk <sileht@mergify.io>
```
;)

File: mergify_engine/tests/functional/test_engine_v2.py [hunk 4]
```diff
@@ -1395,6 +1395,33 @@ def test_refresh_on_conflict(self):
...
+        p1, _ = self.create_pr(files={"TESTING2": "foobar"})
+        p2, _ = self.create_pr(files={"TESTING3": "foobar"})
+        p1.merge()
+
+        self.wait_for("pull_request", {"action": "closed"})
+
+        self.create_message(p2, "@mergifyio update")
+
+        if base.RECORD:
+            time.sleep(1)
```
Comment (jd) [inline mergify_engine/tests/functional/test_engine_v2.py]: looks a bit… fuzzy but ok I guess?

Comment (sileht) [reply mergify_engine/tests/functional/test_engine_v2.py]: I copy/paste the rebase test, but it's not needed here.

**Pull request merged**